Modules/Common/include/Common/TH2Ratio.inl

```diff
@@ -19,19 +19,38 @@
 namespace o2::quality_control_modules::common
 {
 
+template<class T>
+TH2Ratio<T>::TH2Ratio()
+  : T(),
+    o2::mergers::MergeInterface(),
+    mUniformScaling(true)
+{
+  {
+    TDirectory::TContext ctx(nullptr);
+    mHistoNum = new T("num", "num", 10, 0, 10, 10, 0, 10);
+    mHistoDen = new T("den", "den", 1, -1, 1, 1, -1, 1);
+  }
+
+  init();
+}
+
 template<class T>
 TH2Ratio<T>::TH2Ratio(TH2Ratio const& copymerge)
-  : T(copymerge.GetName(), copymerge.GetTitle(),
-         copymerge.getNum()->GetXaxis()->GetNbins(), copymerge.getNum()->GetXaxis()->GetXmin(), copymerge.getNum()->GetXaxis()->GetXmax(),
-         copymerge.getNum()->GetYaxis()->GetNbins(), copymerge.getNum()->GetYaxis()->GetXmin(), copymerge.getNum()->GetYaxis()->GetXmax()),
+  : T(),
     o2::mergers::MergeInterface(),
     mUniformScaling(copymerge.hasUniformScaling())
 {
-  Bool_t bStatus = TH1::AddDirectoryStatus();
-  TH1::AddDirectory(kFALSE);
-  mHistoNum = (T*)copymerge.getNum()->Clone();
-  mHistoDen = (T*)copymerge.getDen()->Clone();
-  TH1::AddDirectory(bStatus);
+  // do not add cloned histograms to gDirectory
+  {
+    TString name_num = T::GetName() + TString("_num");
+    TString name_den = T::GetName() + TString("_den");
+    TString title_num = T::GetTitle() + TString(" num");
+    TString title_den = T::GetTitle() + TString(" den");
+    TDirectory::TContext ctx(nullptr);
+    mHistoNum = new T(name_num, title_num, 10, 0, 10, 10, 0, 10);
+    mHistoDen = new T(name_den, title_den, 1, -1, 1, 1, -1, 1);
+  }
+  copymerge.Copy(*this);
```

Comment:
why allocate these two histograms if, as I assume, they are overwritten by the `Copy` soon after?

Reply:
Because the `Copy()` function assumes that the histograms are already initialized. There is maybe a little overhead in creating dummy histograms and then re-initializing them properly in the `Copy()` function, but on the other hand this minimizes the code duplication between the copy-constructor and the real `Copy()` function.

Reply:
I see, thanks for explaining!